Изменены несколько файлов

SunnatBakidjanov opened the pull request:
--Добавлены данные в menu в mock.js для появления скролла 
--Добавлен компонент ProgressBar.jsx
--Добавлены стили progressBar.module.scss для ProgressBar 
--Добавлена логика useProgressBar.js для ProgressBar 
--Добавлены стили restaurants.module.scss для Restaurants 
--Добавлен файл сброса стилей reset.scss в app
--Добавлен файл mian.scss для общих стилей в app

SunnatBakidjanov committed:
Изменены несколько фалов

--Добавлены данные в menu в mock.js для появления скролла
--Добавлен компонент ProgressBar.jsx
--Добавлены стили progressBar.module.scss для ProgressBar
--Доабвлена логика useProgressBar.js для ProgressBar
--Добавлены стили restaurants.module.scss для Restaurants
--Добавлен файл сброса стилей reset.scss в app
--Добавлен файл mian.scss для общих стилей в app

zubkov7 commented on src/components/app/App.jsx:
```diff
@@ -1,10 +1,14 @@
-        <Layout>
-            <Restaurants />
-        </Layout>
+        <div className='page'>
```
давайте унесем в Layout

zubkov7 replied:
и применим стиль через css модуль

zubkov7 commented on src/components/app/reset.scss:
```diff
@@ -0,0 +1,7 @@
+* {
+    margin: 0;
+    padding: 0;
+    list-style: none;
```
это специфичный стиль для списка, по * плохой подход стилизовать, если нужно сбросить, то нужно указать конкретный тег

zubkov7 commented on src/components/layout/Layout.jsx:
```diff
@@ -1,11 +1,15 @@
+            <ProgressBar />
             <Header />
-            <main className='content'>{children}</main>
+            <main className='content'>
+                <div className='container'>{children}</div>
```
стилизовать необходимо через css модуль

zubkov7 commented on src/components/progressBar/useProgressBar.js:
```diff
@@ -0,0 +1,27 @@
+
+        function progress() {
+            if (progressBar.current && page.current) {
+                const scrollHeight = page.current.scrollHeight - page.current.clientHeight;
+                const scrollTop = page.current.scrollTop;
+                const progressWidth = (scrollTop / scrollHeight) * 100;
```
давайте вынесем вычисление прогресса в отдельную функцию и вынесем из рендера компонента

zubkov7 commented on src/components/progressBar/useProgressBar.js:
```diff
@@ -0,0 +1,27 @@
+                const progressWidth = (scrollTop / scrollHeight) * 100;
+                console.log(progressWidth);
+
+                progressBar.current.style.setProperty("--progress-bar-width", `${progressWidth}%`);
```
для прогресса нужно сделать отдельное состояние и динамически применять через prop style

SunnatBakidjanov committed:
'Внесены изменения'

--div 'page' вынесен в layout и применены стили через модули
--Стиль для списка вынесен отдельно для списка в reset.css
--Элементы в layout стилизованы через модули css
---useProgressBar---
--Вынесены вычесления прогресса в отдельную функци и вынесена из рендера
--Для прогресса сделано отдельное состояние и динамически применено через style

zubkov7 commented on src/components/layout/layout.module.scss:
```diff
@@ -0,0 +1,7 @@
+    background-color: wheat;
+}
+
+.content {}
+
+.container {}
```
пустые селекторы лучше удалять

zubkov7 commented on src/components/progressBar/ProgressBar.jsx:
```diff
@@ -0,0 +1,12 @@
+
+    return (
+        <div className={style.container}>
+            <div className={style.progress} style={{ width: `${progress}%` }}></div>
```
можно использовать сразу закрывающий тег `<div />`
